authentication_strategy_doordash throwing mypy error

### Bug Description

mypy is currently throwing a mypy error on `authentication_strategy_doordash.py`

```console
src/fides/api/ops/service/saas_request/override_implementations/authentication_strategy_doordash.py:50: error: Argument 2 to "assign_placeholders" has
incompatible type "Optional[Any]"; expected "Dict[str, Any]"  [arg-type]
                    "iss": assign_placeholders(self.developer_id, secrets),
                                                                  ^~~~~~~
src/fides/api/ops/service/saas_request/override_implementations/authentication_strategy_doordash.py:51: error: Argument 2 to "assign_placeholders" has
incompatible type "Optional[Any]"; expected "Dict[str, Any]"  [arg-type]
                    "kid": assign_placeholders(self.key_id, secrets),
```

#### Steps to Reproduce

1. Pull the latest main
1. Run mypy

### Expected behavior

No errors.

### Screenshots

_If applicable, add screenshots to help explain your problem._

### Environment

* Version:
* OS:
* Python Version:
* Docker Version:

### Additional context

The PR that added the door dash authentication strategy and a PR to update mypy both landed at about the same time, and because of this the doordash code passed the original mypy, but is failing with the new settings.

--- a/src/fides/api/ops/service/saas_request/override_implementations/authentication_strategy_doordash.py
+++ b/src/fides/api/ops/service/saas_request/override_implementations/authentication_strategy_doordash.py
@@ -1,5 +1,6 @@
 import math
 import time
+from typing import Any, Dict, Optional
 
 import jwt.utils
 from requests import PreparedRequest
@@ -42,13 +43,15 @@ class DoordashAuthenticationStrategy(AuthenticationStrategy):
         Generate a Doordash JWT and add it as bearer auth
         """
 
-        secrets = connection_config.secrets
+        secrets: Optional[Dict[str, Any]] = connection_config.secrets
 
         token = jwt.encode(
             {
                 "aud": "doordash",
-                "iss": assign_placeholders(self.developer_id, secrets),
-                "kid": assign_placeholders(self.key_id, secrets),
+                "iss": assign_placeholders(self.developer_id, secrets)
+                if secrets
+                else None,
+                "kid": assign_placeholders(self.key_id, secrets) if secrets else None,
                 "exp": str(math.floor(time.time() + 60)),
                 "iat": str(math.floor(time.time())),
             },

